src/malcolmWidgets/method/methodPlot.component.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withTheme } from '@material-ui/core/styles';
+import Plot from 'react-plotly.js';
+import { AlarmStates } from '../attributeDetails/attributeAlarm/attributeAlarm.component';
+
+const alarmStatesByIndex = [
+  AlarmStates.NO_ALARM,
+  AlarmStates.MINOR_ALARM,
+  AlarmStates.MAJOR_ALARM,
+  AlarmStates.UNDEFINED_ALARM,
+];
+
+const updatePlotData = (oldDataElement, alarmIndex, attribute) => {
+  const dataElement = oldDataElement;
+  const alarms = attribute.alarmState;
+  dataElement.x = attribute.timeStamp;
+  dataElement.y = attribute.value.map(
+    (value, valIndex) =>
+      alarms[valIndex] === alarmStatesByIndex[alarmIndex] ||
+      (alarms[valIndex - 1] === alarmStatesByIndex[alarmIndex] &&
+        alarms[valIndex] !== AlarmStates.UNDEFINED_ALARM)
+        ? value
+        : null
+  );
+  dataElement.x.push(dataElement.x.slice(-1)[0]);
+  dataElement.y.push(dataElement.y.slice(-1)[0]);
+  dataElement.visible = dataElement.y.some(val => val !== null);
+  return dataElement;
+};
+
+class MethodPlot extends React.Component {
+  static getDerivedStateFromProps(props, state) {
+    const { data, layout } = state;
+    const newData = data.map((dataElement, index) =>
+      updatePlotData(dataElement, index, props.attribute)
+    );
+    layout.datarevision += 1;
+    if (
+      props.attribute.parent !== layout.parent ||
+      props.attribute.name !== layout.attribute
+    ) {
+      layout.parent = props.attribute.parent;
+      layout.attribute = props.attribute.name;
+      layout.yaxis = {
+        color: props.theme.palette.text.primary,
+      };
+    }
+    return {
+      ...state,
+      data: [...newData],
+      layout,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: [
+        {
```

Comment:
this data array construction could also do with a function because of the amount of repetition.